docs/02-for-app-authors/02-requirements.md

```diff
@@ -30,6 +33,87 @@ Every repository must contain a manifest describing how to build the application
 
 This manifest may import other manifest files in the same repository. It's quite common to add the [shared-modules](shared-modules) repository as a submodule.
 
+#### Example manifest
+
+This assumes that upstream brings appdata, icons and desktop files. So we will `install` them. If not, you can add them to your repo and and add them in the `sources` section.
+
+<Tabs groupId="manifest-language" defaultValue="json" queryString>
+  <TabItem value="json" label="json" default>
+
+```json title="io.github.example.MyCoolApp.json"
+{
+  "id": "io.github.example.MyCoolApp",
+  "runtime": "org.freedesktop.Platform",
+  "runtime-version": "23.08",
+  "sdk": "org.freedesktop.Sdk",
+  "command": "mycoolapp",
+  "finish-args": [
+    "--socket=fallback-x11",
+    "--socket=wayland",
+    "--socket=pulseaudio",
+    "--share=network",
+    "--share=ipc"
+  ],
+  "modules": [
+    {
+      "name": "mycoolapp",
+      "buildsystem": "simple",
+      "build-commands": [
+        "install -Dm644 mycoolapp/io.github.example.MyCoolApp.appdata.xml /app/share/appdata/io.github.example.MyCoolApp.appdata.xml",
```

Comment:
The issue here is that you put something that look like a source tarball and the manifest look like it's extracting binaries.

This is really the anti-pattern here.

Reply:
So what's the suggestion, changing the build system to `cmake-ninja`? But that might be considered `too complex`?

Reply:
no. It's actually simpler because all the files that should be upstream are installed by upstream install procedure. You set buildsystem to `meson` :-)

Reply:
Which also means dropping `build-commands`, I'll do that, but I'm unsure, if that would still be helpful for most newcomers. Let's hope they all have meson compatible projects.

Reply:
there is no one-size-fit-all.

Reply:
Agreed, so what's one size fits most?

Reply:
`simple` if we want to encourage binaries, `meson` otherwise, I'd say.

Reply:
Let's go with `meson` then and ignore non oss apps for this part of the docs.

Reply:
There are also many Apps, which don't use a C/C++ buildsystem, so they need to use simple

Reply:
I use meson for Rust, so there.

Again there is no one size-fit-all but the example here was inconsistent.

My problem here is lot of submission comes with "I copied this from documentation" as a justification to push back on reviewers.

Reply:
I just think the simple buildsystem should be showcases, as it's often used

Reply:
I think I'll side with hub here - we probably can't/won't write an example, that allows you to not look at the flatpak docs. So we should strive for a recommended workflow here. But I can probably write some comment, hinting at the existence of `simple`